ext/etc/etc.c

```diff
@@ -518,7 +605,7 @@ etc_getgrnam(VALUE obj, VALUE nam)
     struct group *grp;
     const char *p = StringValueCStr(nam);
 
-    grp = getgrnam(p);
+    grp = mEtc_mutex_synchronize(nogvl_getgrnam, (void *)p);
```

Comment:
As mentioned on Slack, this feels brittle as `nam` could GC while `nogvl_getgrnam` is running and e.g. cause a segfault.
The rb_raise() below might save it due to using `nam`, but that's brittle, so I think `RB_GC_GUARD(nam)` should be used here.

Reply:
I included this in a follow up commit.